core/web/resolver/vrf.go

```diff
@@ -0,0 +1,139 @@
+package resolver
+
+import (
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/vrfkey"
+)
+
+type VRFKeyResolver struct {
+	key vrfkey.KeyV2
+}
+
+func NewVRFKeyResolver(key vrfkey.KeyV2) VRFKeyResolver {
+	return VRFKeyResolver{key}
+}
+
+// ID returns the ID of the VRF key, which is the public key.
+func (k VRFKeyResolver) ID() graphql.ID {
+	return graphql.ID(k.key.PublicKey.String())
+}
+
+func (k VRFKeyResolver) Compressed() string {
+	return k.key.PublicKey.String()
+}
+
+func (k VRFKeyResolver) Uncompressed() string {
+	uncompressed, err := k.key.PublicKey.StringUncompressed()
+	if err != nil {
+		// TODO: figure out better approach
```

Comment:
I think we either figure out how to avoid panics here, or you could just do this before calling `NewVRFKeyResolver` and create a struct that already has this value figured out before reaching this mapping functionality.

The general idea is that `VRFKeyResolver` would not receive a `vrfkey.KeyV2`, but a type that it is similar, but has already handled this error before getting to the resolver.

thoughts @jkongie

Reply:
I've elected to take the somewhat simpler approach and return an error string in this case.

Main reason being: this sort of error popping up _should_ be really rare in reality. Really, if it does pop up, I'm pretty sure the whole app would be borked because the crypto libraries will be returning errors for everything, and the GQL API would probably be the least of our worries. But I agree that we definitely shouldn't panic here.

The reason against having an intermediate type that I can see is that we'd have to create that type somewhere. Right now, the `VRFKeyResolver` is created from the other resolvers. We could have a private type that basically processes the `vrfkey.KeyV2` and extracts the necessary info (id, hash, compressed, uncompressed) and put that in `VRFKeyResolver` instead, but we'd have to handle errors for that private type, and I think we may just end up setting error strings in that case anyways, because the current API doesn't allow for there to be errors in the construction of resolvers.

Let me know what you think!